src/main/scala/circt/stage/phases/CIRCT.scala

```diff
@@ -136,39 +122,65 @@ class CIRCT extends Phase {
-           )
-         })
+        ((circtOptions.target, split) match {
+          case (Some(CIRCTTarget.FIRRTL), false)        => Seq("-ir-fir")
+          case (Some(CIRCTTarget.HW), false)            => Seq("-ir-hw")
+          case (Some(CIRCTTarget.Verilog), true)        => Seq("--split-verilog")
+          case (Some(CIRCTTarget.Verilog), false)       => None
+          case (Some(CIRCTTarget.SystemVerilog), true)  => Seq("--split-verilog")
+          case (Some(CIRCTTarget.SystemVerilog), false) => None
+          case (None, _)                                =>
+            throw new Exception(
+              "No 'circtOptions.target' specified. This should be impossible if dependencies are satisfied!"
+            )
+          case _                                        =>
+            throw new Exception(
+              s"Invalid combination of circtOptions.target ${circtOptions.target} and split ${split}"
+            )
+        })
 
     try {
       logger.info(s"""Running CIRCT: '${cmd.mkString(" ")} < $$input'""")
-      val result = (cmd #< new java.io.ByteArrayInputStream(input.getBytes)).!!
-
-      circtOptions.target match {
-        case Some(CIRCTTarget.FIRRTL) =>
-          Seq(EmittedMLIR(outputFileName, result, Some(".fir.mlir")))
-        case Some(CIRCTTarget.HW) =>
-          Seq(EmittedMLIR(outputFileName, result, Some(".hw.mlir")))
-        case Some(CIRCTTarget.Verilog) =>
-          Seq(EmittedVerilogCircuitAnnotation(EmittedVerilogCircuit(outputFileName, result, ".v")))
-        case Some(CIRCTTarget.SystemVerilog) =>
-          Seq(EmittedVerilogCircuitAnnotation(EmittedVerilogCircuit(outputFileName, result, ".sv")))
-        case None => throw new Exception(
-          "No 'circtOptions.target' specified. This should be impossible if dependencies are satisfied!"
-        )
+      val stdoutStream = new java.io.ByteArrayOutputStream
+      val stderrStream = new java.io.ByteArrayOutputStream
+      val stdoutWriter = new java.io.PrintWriter(stdoutStream)
+      val stderrWriter = new java.io.PrintWriter(stderrStream)
+      val exitValue    = (cmd #< new java.io.ByteArrayInputStream(input.getBytes)).!(ProcessLogger(stdoutWriter.println, stderrWriter.println))
+      stdoutWriter.close()
+      stderrWriter.close()
+      val result       = stdoutStream.toString
+      logger.info(result)
+      val errors       = stderrStream.toString
+      if (exitValue != 0) {
+        //logger.error(errors)
+        StageUtils.dramaticError(s"${binary} failed.\nExitCode:\n${exitValue}\nSTDOUT:\n${result}\nSTDERR:\n${errors}")
+        throw new StageError()
+      }
+      if (split) { 
+        logger.info(result)
+        Nil
+      } else {// if split it has already been written out to the file system stdout not necessary.
+        val outputFileName: String                   = stageOptions.getBuildFileName(firrtlOptions.outputFileName.get)
```

Comment:
other functional change, only write out via stdout if --split-verilog was not already sent to firtool